autobotAI_integrations/integrations/aws_bedrock/__init__.py

```
@@ -317,11 +317,38 @@ def get_pydantic_agent(
 
         if not credentials:
             credentials = self._temp_credentials()
 
         model_instance = self.get_pydantic_model(model,credentials=credentials)
+
+        generic_settings = options.pop("model_settings", {})
+        provider_settings = {}
+
+        t = generic_settings.get("temperature")
+        if t is not None:
+            if not bedrock_model_rejects_temperature(model):
```

Ruff (0.15.21)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)